core/src/main/java/cz/xtf/core/waiting/helpers/FailFastBoolSupplierBuilder.java

```diff
@@ -0,0 +1,149 @@
+package cz.xtf.core.waiting.helpers;
+
+
+import cz.xtf.core.config.BuildManagerConfig;
+import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.event.EventList;
+import cz.xtf.core.event.EventListFilter;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+public class FailFastBoolSupplierBuilder {
```

Comment:
This class needs documentation. What is it's purpose? How do I use it? It's very abstract.

Reply:
Please check.

Reply:
Good documentation, some grammar-related nitpicks:

* {@link BooleanSupplier} that check -> {@link BooleanSupplier} that checks
* due to a error state of cluster -> due to an error state of the cluster
* returns true, waiter should fail -> returns true, the waiter should fail
* returns true, whole fail fast -> returns true, the whole fail fast

Reply:
LGTM.